src/Microsoft.AspNetCore.Mvc.Core/ApiExplorer/ApiConventionResult.cs

```diff
@@ -26,31 +26,38 @@ public ApiConventionResult(IReadOnlyList<IApiResponseMetadataProvider> responseM
             ApiConventionTypeAttribute[] apiConventionAttributes,
             out ApiConventionResult result)
         {
-            foreach (var attribute in apiConventionAttributes)
+            var apiConventionMethodAttribute = method.GetCustomAttribute<ApiConventionMethodAttribute>(inherit: true);
+            var conventionMethod = apiConventionMethodAttribute?.Method;
+            if (conventionMethod == null)
             {
-                var conventionMethod = GetConventionMethod(method, attribute.ConventionType);
-                if (conventionMethod != null)
-                {
-                    var metadataProviders = conventionMethod.GetCustomAttributes(inherit: false)
-                        .OfType<IApiResponseMetadataProvider>()
-                        .ToArray();
+                conventionMethod = GetConventionMethod(method, apiConventionAttributes);
+            }
 
-                    result = new ApiConventionResult(metadataProviders);
-                    return true;
-                }
+            if (conventionMethod != null)
```

Comment:
So we're saying in this case that a convention method takes precedence over `IApiResponseMP` - is that because lots of things implement that interface other than `ProducesReponseType`?

Seems right to me, just poking it.

Reply:
> So we're saying in this case that a convention method takes precedence over IApiResponseMP

Actually no. Looking up attributes on the actual method continues to happen independently as part of ApiExplorer. All we do here is lookup metadata inferred from API conventions. Here's the precedence order:

a) `IApiResponseMetadataProvider` specified on an action
b) `ApiConventionMethodAttribute`
c) `ApiConventionTypeAttribute` on controller
d) `ApiConventionTypeAttribute` on assembly